### PR TITLE
prefix for gemfire configuration properties - common, source and sink

### PR DESCRIPTION
**Pull request by sobychacko**

### CODE REVIEW

File: gemfire/spring-cloud-starter-stream-source-gemfire/src/main/resources/gemfire-source.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #
 # Gemfire client pool must be configured with subscriptions enabled
 #
-subscriptionEnabled:true
+gemfire:
+    pool:
+        subscriptionEnabled:true
```
Comment (artembilan) [inline gemfire/spring-cloud-starter-stream-source-gemfire/src/main/resources/gemfire-source.yml]: We can't use `yml` for such a cases: http://docs.spring.io/spring-boot/docs/1.4.0.BUILD-SNAPSHOT/reference/html/boot-features-external-config.html#boot-features-external-config-yaml-shortcomings.

Not sure if we can overcome it as default somehow from the code, but at least right now we have to convert it into `.properties`.

File: gemfire/spring-cloud-starter-stream-sink-gemfire/src/test/java/org/springframework/cloud/stream/app/gemfire/sink/GemfireSinkIntegrationTests.java [hunk 2]
```diff
@@ -37,24 +48,16 @@
...
 /**
  * @author David Turanski
  */
 @RunWith(SpringJUnit4ClassRunner.class)
 @SpringApplicationConfiguration(classes = {GemfireSinkIntegrationTests.GemfireSinkApplication.class})
-@IntegrationTest({"regionName=Stocks", "keyExpression='key'", "hostAddresses=localhost:42424", "connectType=server"})
+@IntegrationTest({"gemfire.region.regionName=Stocks", "gemfire.keyExpression='key'", "gemfire.pool.hostAddresses=localhost:42424",
+		"gemfire.pool.connectType=server"})
 @EnableConfigurationProperties(GemfireSinkProperties.class)
 public class GemfireSinkIntegrationTests {
```
Comment (artembilan) [inline gemfire/spring-cloud-starter-stream-sink-gemfire/src/test/java/org/springframework/cloud/stream/app/gemfire/sink/GemfireSinkIntegrationTests.java:62]: This test works for me, but I don't like that explicit `port` for embedded Gemfire.
Although looks like we can't have a random one there...

Anyway looks like that https://github.com/spring-cloud/spring-cloud-stream-modules/issues/49 is most like some race condition during connection. Or in Spring Data Gemfire or in the Gemfire per se.

After upgrading to SDG-1.8.1.RELEASE and, therefore, Gemfire 8.2.0 in SI, looks like we have stopped to fail in our CI plans for similar reason. 

Not sure if we should upgrade here to SD-Hopper right now, but at least we should keep that in mind for the future, e.g. after switching to Boot-1.4.

**Comment by artembilan** [comment]
Both `Source` tests fails for me with:

```
Caused by: com.gemstone.gemfire.cache.NoSubscriptionServersAvailableException: Primary discovery failed.
    at com.gemstone.gemfire.cache.client.internal.QueueManagerImpl$ConnectionList.setPrimaryDiscoveryFailed(QueueManagerImpl.java:1300)
    at com.gemstone.gemfire.cache.client.internal.QueueManagerImpl.initializeConnections(QueueManagerImpl.java:434)
    at com.gemstone.gemfire.cache.client.internal.QueueManagerImpl.start(QueueManagerImpl.java:302)
    at com.gemstone.gemfire.cache.client.internal.PoolImpl.start(PoolImpl.java:322)
    at com.gemstone.gemfire.cache.client.internal.PoolImpl.finishCreate(PoolImpl.java:150)
    at com.gemstone.gemfire.cache.client.internal.PoolImpl.create(PoolImpl.java:136)
    at com.gemstone.gemfire.internal.cache.PoolFactoryImpl.create(PoolFactoryImpl.java:448)
    at org.springframework.data.gemfire.client.PoolFactoryBean.afterPropertiesSet(PoolFactoryBean.java:183)
```

Not familiar with that beast.

Maybe worth to include John Blum? 
At least for some consultation on the matter :smile: .

File: gemfire/spring-cloud-starter-stream-source-gemfire-cq/src/main/java/org/springframework/cloud/stream/app/gemfire/cq/source/GemfireCqSourceProperties.java [hunk 3]
```diff
@@ -25,7 +25,7 @@
  *
  * @author David Turanski
  */
-@ConfigurationProperties
+@ConfigurationProperties("gemfire.cq")
```
Comment (artembilan) [inline gemfire/spring-cloud-starter-stream-source-gemfire-cq/src/main/java/org/springframework/cloud/stream/app/gemfire/cq/source/GemfireCqSourceProperties.java]: I think this is fine to be as just `gemfire` as well.
It's enough odd to have to `source` modules in the same stream or even aggregate app.
Or am I missing anything in this prefixes woods?

Comment (sobychacko) [inline gemfire/spring-cloud-starter-stream-source-gemfire-cq/src/main/java/org/springframework/cloud/stream/app/gemfire/cq/source/GemfireCqSourceProperties.java]: that makes sense - i was worried about collisions, but I think we are good here.

Comment (sobychacko) [inline gemfire/spring-cloud-starter-stream-source-gemfire-cq/src/main/java/org/springframework/cloud/stream/app/gemfire/cq/source/GemfireCqSourceProperties.java]: committed the fix.

File: gemfire/spring-cloud-starter-stream-source-gemfire/src/main/resources/gemfire-source.yml (hunk 1 above)
Comment (artembilan) [inline gemfire/spring-cloud-starter-stream-source-gemfire/src/main/resources/gemfire-source.yml]: How about this concern?
It is really bug. I won't pass PR without the fix! :smile:

Comment (sobychacko) [inline gemfire/spring-cloud-starter-stream-source-gemfire/src/main/resources/gemfire-source.yml]: This is addressed now.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-cloud/spring-cloud-stream-app-starters/commit/8b4540cae9b1e85277241547d306cfeb5dc8b372

**Pull request closed without merging**